5.analyze_data/scripts/0.generate_UMAP_emeddings.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Generate UMAP embeddings per plate and profile (single-cell or organoid)
+
+# In[1]:
+
+
+import glob
+import pathlib
+import pandas as pd
+import umap
+
+from pycytominer import feature_select
+from pycytominer.cyto_utils import infer_cp_features
+
+
+# ## Set paths and constants
+
+# In[2]:
+
+
+# Set constants
+umap_random_seed = 0
+umap_n_components = 2
+
+output_dir = pathlib.Path("results")
+output_dir.mkdir(parents=True, exist_ok=True)
+
+
+# ## Identify file paths to process
+
+# In[3]:
+
+
+# Set input paths
+data_dir = pathlib.Path("../4.preprocess_features/data/single_cell_profiles")
+
+# Select only the feature selected files
+file_suffix = "*feature_selected.parquet"
+
+# Obtain file paths for all feature selected plates
+fs_files = glob.glob(f"{data_dir}/{file_suffix}")
+fs_files
+
+
+# In[4]:
+
+
+# Load feature data into a dictionary, keyed on plate name
+cp_dfs = {x.split("/")[-1]: pd.read_parquet(x) for x in fs_files}
+
+# Print out useful information about each dataset
+print(cp_dfs.keys())
+[cp_dfs[x].shape for x in cp_dfs]
+
+
+# In[5]:
+
+
+desired_columns = [
+    "Metadata_Plate",
+    "Metadata_Well",
+    "Metadata_Site",
+    "Metadata_treatment",
+    "Metadata_dose",
+    "Metadata_ZSlice",
+    "Metadata_Nuclei_Location_Center_X",
+    "Metadata_Nuclei_Location_Center_Y",
+]
+
+# Fit UMAP features per dataset and save
+for plate in cp_dfs:
+    # Extract the first two parts of the plate name
+    plate_name_parts = pathlib.Path(plate).stem.split("_")[:2]
+    plate_name = "_".join(plate_name_parts)
+    print("UMAP embeddings being generated for", plate_name)
+
+    # Set compartments based on the second part of the plate name
+    if plate_name_parts[1] == "sc":
+        compartments = ["nuclei", "cells", "cytoplasm"]
+    elif plate_name_parts[1] == "organoid":
+        compartments = ["organoids"]
+    else:
+        compartments = []  # or handle other cases if necessary
```

Comment:
Could consider doing something like this again instead:
`compartments = ["nuclei", "cells", "cytoplasm"] if profile == "sc" else ["organoid"]`